webapp/src/components/MapShowLocations/index.js

```diff
@@ -5,24 +5,151 @@ import mapboxgl from 'mapbox-gl'
 import MapboxGeocoder from '@mapbox/mapbox-gl-geocoder'
 import Box from '@material-ui/core/Box'
 import { useQuery } from '@apollo/react-hooks'
+import { makeStyles } from '@material-ui/styles'
+import Grid from '@material-ui/core/Grid';
 
-import { mapboxConfig } from '../../config'
+import InputAdornment from '@material-ui/core/InputAdornment';
+import Checkbox from '@material-ui/core/Checkbox';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+import TextField from '@material-ui/core/TextField';
+import Divider from '@material-ui/core/Divider';
+import Button from '@material-ui/core/Button';
+import Modal from '@material-ui/core/Modal';
+import IconButton from '@material-ui/core/IconButton';
+import MenuIcon from '@material-ui/icons/Menu';
+
+import { mapboxConfig, constants } from '../../config'
 import MapMarker from '../MapMarker'
 import MapPopup from '../MapPopup'
 import { GET_NEARBY_LOCATIONS_QUERY } from '../../gql'
 
+
 const initialZoom = 12.5
-const distance = 5000
+var map = null
+var checkLifebank = true
+var checkSponsor = true
+var markerList = []
+var searchDistance = 5000
+var filterSponsorcategory = "All"
+
+const useStyles = makeStyles((theme) => ({
+  mapOverlay: {
+    position: "absolute",
+    top: 110,
+    '@media only screen and (max-width: 900px)': {
+      top: 120,
+    },
+    right: 10,
+    boxShadow: "0 1px 5px rgba(0, 0, 0, 0.2)"
+  },
+  mapOverlayinner: {
+    backgroundColor: "#fff",
+    borderRadius: "3px",
+    padding: "5px",
+  },
+  icon: {
+    fontSize: 18,
+  },
+  paper: {
+    position: 'absolute',
+    width: 400,
+    backgroundColor: "#FFF",
+    boxShadow: theme.shadows[5],
+    padding: theme.spacing(2, 4, 3),
+  },
+  inputStyle: {
+    width: "100%",
+    marginBottom: 15
+  },
+  checkBoxForm: {
+    marginBottom: 10
+  },
+  buttonStyle: {
+    width: "100%",
+    marginBottom: 15
+  },
+}))
+
+var customData = {
+  'features': [],
+  'type': 'FeatureCollection'
+};
+
+const defaultCategories = [
+  "All"
+]
+
+const {
+  SPONSOR_TYPES
+} = constants
+
+const sponsorsCategories = defaultCategories.concat(SPONSOR_TYPES)
+
+function getModalStyle() {
+  return {
+    top: `50%`,
+    left: `50%`,
+    transform: `translate(-50%, -50%)`,
+  };
+}
 
 function MapShowLocations({ location, ...props }) {
+  const [distance, setDistance] = React.useState(5000);
+  const [state, setState] = React.useState({
+    checkedLifebank: true,
+    checkedSponsor: true,
+  });
+  const [modalStyle] = React.useState(getModalStyle);
+  const [open, setOpen] = React.useState(false);
+  const [sponsorsCat, setSponsorsCat] = React.useState('All');
+
+  const handleChange = (event) => {
+    setState({ ...state, [event.target.name]: event.target.checked });
+    if (event.target.name === "checkedLifebank") {
```

Comment:
add blank space after the `setState`

Also we can do better that logic removing the IF STATEMENT and using SWITCH, if you want to switch the boolean values try to do something like that:
```
switch(name) {
  case 'checkedLifebank': {
    checkLifebank = !checkLifebank
    
    break
  }
    
  default: {
  checkSponsor = !checkSponsor
    
    break
  }
}
```